### PR TITLE
Feat: Add Profile Screen

### LINKED ISSUE #35: Feat: Add support for Deleting Account from Profile Screen
Need to add a feature of deleting account along with the user data to adhere the Goggle Play Store policies.

### LINKED ISSUE #37: (feat) Add support for account details page and deleting the account
- Add a section where user is able to see account details
- Add support for letting the user to delete their account

### PR DESCRIPTION
**Pull request by abhishek310001**
## Description
_This PR is introduced to add the feature of updating, deleting the user's account by introducing `Profile Screen`._

## Screenshots

| iOS | Android |
| ---- | ---- |
| Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 18 39 52 | Screenshot_1722258649
Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 18 39 59 | Screenshot_1722258652
Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 18 40 04 | Screenshot_1722258657
Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 18 40 09 | Screenshot_1722258660
Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 18 40 14 | Screenshot_1722258665

## Video
_Video demonstrating the usage of new feature and UI._

https://github.com/user-attachments/assets/e9128793-869b-4f6c-acbe-a771aacbdc95

### Automated test cases added
NA

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._
TBA

### Devices Tested
- Google Pixel 7 (Android 14)
- iPhone 15 Pro (iOS 17.5)

### CODE REVIEW

**Comment by bbbneo333** [comment]
@abhishek310001 - Looks good, but here are couple of things that we should look into:
- The most frequently used layouts, in both iOS and Android - They both uses a navigation bar which is fixed to the bottom of the screen. Checkout apps for Swiggy/Zomato or see following SS:

Screenshot 2024-07-25 at 19 20 23
Screenshot 2024-07-25 at 19 23 47

Bottom line - The drawer layout is outdated is not in use in latest apps. We should add support for the new layout. We can just have two sections - Home and Account with suitable icons.

- The gap between the screen edges still seems pretty high - Anything we can do about that? Basically - Make sure these margins matches (see SS). As of now - I think they don't.

352123398-138e4cf6-1df9-4498-858a-e4a080312ce3

- The "Delete Account" button seems to be _hanging_ - Can we find a new placement for it please?

**Comment by bbbneo333** [comment]
UI looks clean now!
One final note - The gap seems to be too much around the corners in bottom nav bar for iOS. The Android one looks good though. Anything we can do about that?

Screenshot 2024-07-30 at 12 00 41

**Comment by abhishek310001** [comment]
> UI looks clean now! One final note - The gap seems to be too much around the corners in bottom nav bar for iOS. The Android one looks good though. Anything we can do about that?
> 
> Screenshot 2024-07-30 at 12 00 41

This is due to the safeArea inset provided by the device itself if we want to reduce the gap we can ignore the bottom safeArea while maintaining the top safeArea. It will look like this if we do that.
| Simulator Screenshot - iPhone 15 Pro - 2024-07-30 at 14 11 09 | Screenshot_1722328775 |
| ---- | ---- |

File: react-native.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  dependencies: {
+    'react-native-vector-icons': {
+      platforms: {
+        ios: null,
```
Comment (bbbneo333) [inline react-native.config.js]: Not sure why this is added, can you add comment in there if this is required?

**Review by bbbneo333 — APPROVED** [review]

File: react-native.config.js (hunk 1 above)
Comment (abhishek310001) [reply react-native.config.js]: Yeah this is required, added a comment stating why it is required.
For more reference you can checkout iOS setup: https://www.npmjs.com/package/react-native-vector-icons

**Review by bbbneo333 — APPROVED** [review]

**Pull request merged**